packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -301,7 +306,7 @@ abstract class VpcBase extends Resource implements IVpc {
   /**
    * Identifier for the VPN gateway
    */
-  public abstract readonly vpnGatewayId?: string;
+  public abstract vpnGatewayId?: string;
```

Comment:
I don't think the public API should be that this value is mutable.

This declaration should remain the same, and on the concrete class (`Vpc`) you would implement it as a getter combined with a mutable private field.

Reply:
Done and added the function obtainVpnGatewayId() but it does not feel right, specifically that in java you will get beside that a getVpnGatewayId()